Crash after momo payment

## Describe the Bug
The app crashes after successful Momo payment.
@jeremiahAI  Please can you confirm this?

Thanks

# error message 

E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.flutterwave.rave_android, PID: 29835
    java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=70876, result=111, data=Intent {  }} to activity {com.flutterwave.rave_android/com.flutterwave.raveandroid.RavePayActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.flutterwave.raveandroid.rave_remote.responses.MobileMoneyChargeResponse$Data.getProvider()' on a null object reference
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4845)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:4886)
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.flutterwave.raveandroid.rave_remote.responses.MobileMoneyChargeResponse$Data.getProvider()' on a null object reference
        at com.flutterwave.raveandroid.rave_presentation.ghmobilemoney.GhMobileMoneyHandler.requeryTx(GhMobileMoneyHandler.java:138)
        at com.flutterwave.raveandroid.rave_presentation.ghmobilemoney.GhMobileMoneyHandler.requeryTx(GhMobileMoneyHandler.java:127)
        at com.flutterwave.raveandroid.ghmobilemoney.GhMobileMoneyFragment.onActivityResult(GhMobileMoneyFragment.java:245)
        at androidx.fragment.app.FragmentActivity.onActivityResult(FragmentActivity.java:170)
        at android.app.Activity.dispatchActivityResult(Activity.java:8110)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4838)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:4886) 
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7356) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930) 

## System Information

 - Operating system: MacOS
 - Rave version: 2.1.25

@jeremiahAI 

This is still not fixed as of version. **2.1.16**

I think it would have been fair to hear my feedback before **closing this issue**.

Please re-open this issue as its not fixed yet.

**Error Message**

`E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.flutterwave.rave_android, PID: 23747
    java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=70876, result=111, data=Intent {  }} to activity {com.flutterwave.rave_android/com.flutterwave.raveandroid.RavePayActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.flutterwave.raveandroid.rave_remote.responses.MobileMoneyChargeResponse$Data.getProvider()' on a null object reference
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4845)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:4886)
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.flutterwave.raveandroid.rave_remote.responses.MobileMoneyChargeResponse$Data.getProvider()' on a null object reference
        at com.flutterwave.raveandroid.rave_presentation.ghmobilemoney.GhMobileMoneyHandler.requeryTx(GhMobileMoneyHandler.java:138)
        at com.flutterwave.raveandroid.rave_presentation.ghmobilemoney.GhMobileMoneyHandler.requeryTx(GhMobileMoneyHandler.java:127)
        at com.flutterwave.raveandroid.ghmobilemoney.GhMobileMoneyFragment.onActivityResult(GhMobileMoneyFragment.java:245)
        at androidx.fragment.app.FragmentActivity.onActivityResult(FragmentActivity.java:170)
        at android.app.Activity.dispatchActivityResult(Activity.java:8110)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4838)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:4886) 
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7356) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930) `

--- a/rave_presentation/src/main/java/com/flutterwave/raveandroid/rave_presentation/ghmobilemoney/GhMobileMoneyHandler.java
+++ b/rave_presentation/src/main/java/com/flutterwave/raveandroid/rave_presentation/ghmobilemoney/GhMobileMoneyHandler.java
@@ -135,7 +135,7 @@ public class GhMobileMoneyHandler implements GhMobileMoneyContract.Handler {
         body.setTx_ref(txRef);
         body.setPBFPubKey(publicKey);
 
-        if (data.getProvider() != null && data.getProvider().equalsIgnoreCase(RaveConstants.eTransact_GH)){
+        if (data != null && data.getProvider() != null && data.getProvider().equalsIgnoreCase(RaveConstants.eTransact_GH)){
             mInteractor.showPollingIndicator(true, data.getValidateInstructions());
         }else{
             mInteractor.showPollingIndicator(true, "");

